java/squeek/appleskin/client/HUDOverlayHandler.java

```diff
@@ -134,68 +151,157 @@ public static void drawSaturationOverlay(MatrixStack matrixStack, float saturati
 		if (saturationLevel + saturationGained < 0)
 			return;
 
-		int startBar = saturationGained != 0 ? Math.max(0, (int) saturationLevel / 2) : 0;
-		int endBar = (int) Math.ceil(Math.min(20, saturationLevel + saturationGained) / 2f);
-		int barsNeeded = endBar - startBar;
+		enableAlpha(alpha);
 		mc.getTextureManager().bindTexture(modIcons);
 
-		enableAlpha(alpha);
-		for (int i = startBar; i < startBar + barsNeeded; ++i)
+		float modifiedSaturation = Math.min(saturationLevel + saturationGained, 20);
+
+		int startSaturationBar = 0;
+		int endSaturationBar = (int)Math.ceil(modifiedSaturation / 2.0F);
+
+		if (saturationGained != 0)
+			startSaturationBar = (int)Math.max(saturationLevel / 2.0F, 0);
+
+		int iconSize = 9;
+
+		for (int i = startSaturationBar; i < endSaturationBar; ++i)
```

Comment:
Why did these lines change? `startSaturationBar` shouldn't always start at 0, that will redraw on top of already filled saturation bars for the flashing saturation. I like the previous version better.

Reply:
When `saturationGained` is not 0, it will be relocated to the correct location.
This not different from the original code, I think the logic is clearer (maybe need some comment).
If you prefer the "?:" version, I'm willing to change it back.

Reply:
Yeah, sorry, I missed the `if (saturationGained != 0)` part.

Reply:
nevermind, I have seen your edited, but since it was `penning` content just now, GitHub will submit when review complete